plugins/modules/user_info.py

```diff
@@ -5,97 +5,59 @@
 
 from __future__ import absolute_import, division, print_function
 
-from typing import Any, List, Optional
-
 import ansible_collections.linode.cloud.plugins.module_utils.doc_fragments.user_info as docs
-from ansible_collections.linode.cloud.plugins.module_utils.linode_common import (
-    LinodeModuleBase,
-)
-from ansible_collections.linode.cloud.plugins.module_utils.linode_docs import (
-    global_authors,
-    global_requirements,
-)
-from ansible_specdoc.objects import (
-    FieldType,
-    SpecDocMeta,
-    SpecField,
-    SpecReturnValue,
+from ansible_collections.linode.cloud.plugins.module_utils.linode_common_info import (
+    InfoModule,
+    InfoModuleAttr,
+    InfoModuleResult,
 )
+from ansible_specdoc.objects import FieldType
 from linode_api4 import User
 
-spec = {
-    # Disable the default values
-    "label": SpecField(type=FieldType.string, required=False, doc_hide=True),
-    "state": SpecField(type=FieldType.string, required=False, doc_hide=True),
-    "username": SpecField(
-        type=FieldType.string,
-        required=True,
-        description=["The username of the user."],
+module = InfoModule(
+    primary_result=InfoModuleResult(
+        field_name="user",
+        field_type=FieldType.dict,
+        display_name="User",
+        docs_url="https://techdocs.akamai.com/linode-api/reference/get-user",
+        samples=docs.result_user_samples,
     ),
-}
-
-SPECDOC_META = SpecDocMeta(
-    description=["Get info about a Linode User."],
-    requirements=global_requirements,
-    author=global_authors,
-    options=spec,
-    examples=docs.specdoc_examples,
-    return_values={
-        "user": SpecReturnValue(
-            description="The user info in JSON serialized form.",
-            docs_url="https://techdocs.akamai.com/linode-api/reference/get-user",
-            type=FieldType.dict,
-            sample=docs.result_user_samples,
-        ),
-        "grants": SpecReturnValue(
-            description="The grants info in JSON serialized form.",
+    secondary_results=[
+        InfoModuleResult(
+            field_name="grants",
+            field_type=FieldType.dict,
+            display_name="Grants",
             docs_url="https://techdocs.akamai.com/linode-api/reference/get-user-grants",
-            type=FieldType.dict,
-            sample=docs.result_grants_samples,
-        ),
-    },
+            samples=docs.result_grants_samples,
+            get=lambda client, user, params: client.get(
+                # We can't use the UserGrants type here because
+                # it does not serialize directly to JSON or store
+                # the API response JSON.
```

Comment:
Do we want to make UserGrants serializable as a Python SDK issue?

Reply:
Good point! Yeah I think that's a good idea